xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp

```diff
@@ -177,102 +182,99 @@ void CGUIDialogVideoManagerVersions::SetDefaultVideoVersion(const CFileItem& ver
 {
-  const int dbId{m_videoAsset->GetVideoInfoTag()->m_iDbId};
-  const MediaType mediaType{m_videoAsset->GetVideoInfoTag()->m_type};
-  const VideoDbContentType itemType{m_videoAsset->GetVideoContentType()};
-
-  // prompt to choose a video file
-  VECSOURCES sources{*CMediaSourceSettings::GetInstance().GetSources("files")};
+  CVideoDatabase videoDb;
+  if (!videoDb.Open())
+  {
+    CLog::LogF(LOGERROR, "Failed to open video database!");
+    return false;
+  }
 
-  CServiceBroker::GetMediaManager().GetLocalDrives(sources);
-  CServiceBroker::GetMediaManager().GetNetworkLocations(sources);
+  CFileItemList items;
+  if (!GetSimilarMovies(m_videoAsset, items, videoDb))
+    return false;
 
-  std::string path;
-  if (CGUIDialogFileBrowser::ShowAndGetFile(
-          sources, CServiceBroker::GetFileExtensionProvider().GetVideoExtensions(),
-          g_localizeStrings.Get(40014), path))
+  if (items.Size() == 0)
   {
-    std::string typeVideoVersion;
-    std::string videoTitle;
-    int idFile{-1};
-    int idMedia{-1};
-    MediaType itemMediaType;
-    VideoAssetType videoAssetType{VideoAssetType::UNKNOWN};
+    // No button = browse library
+    // Yes button = browse files
+    // Custom button = cancel
 
-    const int idVideoVersion{m_database.GetVideoVersionInfo(path, idFile, typeVideoVersion, idMedia,
-                                                            itemMediaType, videoAssetType)};
+    const int dlgResult = CGUIDialogYesNo::ShowAndGetInput(
+        CVariant{40030}, CVariant{40032}, CVariant{40029}, CVariant{40028}, CVariant{222}, 0);
 
-    if (idVideoVersion != -1)
+    switch (dlgResult)
     {
-      CFileItemList versions;
-      m_database.GetVideoVersions(itemType, dbId, versions);
-      if (std::any_of(versions.begin(), versions.end(),
-                      [idFile](const std::shared_ptr<CFileItem>& version)
-                      { return version->GetVideoInfoTag()->m_iDbId == idFile; }))
-      {
-        CGUIDialogOK::ShowAndGetInput(
-            CVariant{40014}, StringUtils::Format(g_localizeStrings.Get(40016), typeVideoVersion));
-        return;
-      }
+      case -1:
+      case 2:
+        // Cancel or Custom/Cancel
+        return false;
 
-      if (itemMediaType == MediaTypeMovie)
+      case 0:
       {
-        videoTitle = m_database.GetMovieTitle(idMedia);
-      }
-      else
-        return;
+        // No = Browse library
+        if (!GetAllOtherMovies(m_videoAsset, items, videoDb))
+          return false;
 
-      if (!CGUIDialogYesNo::ShowAndGetInput(
-              CVariant{40014},
-              StringUtils::Format(g_localizeStrings.Get(40017), typeVideoVersion, videoTitle)))
-      {
-        return;
+        return ChooseVideoAndConvertToVideoVersion(
+            items, m_videoAsset->GetVideoContentType(), m_videoAsset->GetVideoInfoTag()->m_type,
+            m_videoAsset->GetVideoInfoTag()->m_iDbId, videoDb, MediaRole::Parent);
```

Comment:
Real nitpicking: I'm always trying to avoid same function calls in a row if I can. Especially if I know that the implementation is not inline / not a one-liner. Looking at the implementation of `CFileItem::GetVideoInfoTag`, I think we should introduce a variable here instead calling it two times in a row.

```c++
CVideoInfoTag* CFileItem::GetVideoInfoTag()
{
  // Note: CPVRRecording is derived from CVideoInfoTag
  if (m_pvrRecordingInfoTag)
    return m_pvrRecordingInfoTag.get();
  else if (!m_videoInfoTag)
    m_videoInfoTag = new CVideoInfoTag;

  return m_videoInfoTag;
}

const CVideoInfoTag* CFileItem::GetVideoInfoTag() const
{
  // Note: CPVRRecording is derived from CVideoInfoTag
  return m_pvrRecordingInfoTag ? m_pvrRecordingInfoTag.get() : m_videoInfoTag;
}
```

Reply:
Agreed.